test/p2p_c_version/answerer.c

```diff
@@ -0,0 +1,320 @@
+
+ #include <rtc/rtc.h>
+
+ #include <stdbool.h>
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <string.h>
+ #include <unistd.h> // for sleep
+ #include <ctype.h>
+
+typedef struct {
+	rtcState state;
+	rtcGatheringState gatheringState;
+	int pc;
+	int dc;
+	bool connected;
+} Peer;
+
+Peer *peer = NULL;
+static void dataChannelCallback(int dc, void *ptr) ;
+static void descriptionCallback(const char *sdp, const char *type, void *ptr);
+
+static void candidateCallback(const char *cand, const char *mid, void *ptr);
+
+static void stateChangeCallback(rtcState state, void *ptr);
+
+static void gatheringStateCallback(rtcGatheringState state, void *ptr);
+
+// static void openCallback(void *ptr);
+
+static void closedCallback(void *ptr);
+
+static void messageCallback(const char *message, int size, void *ptr);
+
+static void deletePeer(Peer *peer);
+int all_space(const char *str);
+char* state_print(rtcState state);
+char* rtcGatheringState_print(rtcState state);
+
+
+int main(int argc, char **argv) {
+	rtcInitLogger(RTC_LOG_DEBUG);
+
+	// Create peer
+	rtcConfiguration config;
+	memset(&config, 0, sizeof(config));
+
+      Peer *peer = (Peer *)malloc(sizeof(Peer));
+	if (!peer){
+
+            printf("Error allocating memory for peer\n");
+            deletePeer(peer);
+
+      }
+	memset(peer, 0, sizeof(Peer));
+
+      printf("Peer created\n");
+
+      // Create peer connection
+      peer->pc = rtcCreatePeerConnection(&config);
+
+	rtcSetUserPointer(peer->pc, peer);
+	rtcSetLocalDescriptionCallback(peer->pc, descriptionCallback);
+	rtcSetLocalCandidateCallback(peer->pc, candidateCallback);
+	rtcSetStateChangeCallback(peer->pc, stateChangeCallback);
+	rtcSetGatheringStateChangeCallback(peer->pc, gatheringStateCallback);
+
+      rtcSetUserPointer(peer->dc, NULL);
+	rtcSetDataChannelCallback(peer->pc, dataChannelCallback);
+
+
+
+
+	bool exit = false;
+
+	while (!exit) {
+
+		printf("\n");
+		printf("***************************************************************************************\n");
+
+	     // << endl
+	     printf("* 0: Exit /"
+			" 1: Enter remote description /"
+			" 2: Enter remote candidate /"
+			" 3: Send message /"
+			" 4: Print Connection Info *\n"
+			"[Command]: ");
+
+		int command = -1;
+		// int check_scan
+		if (scanf("%d", &command)){
+			
+
+		}else {
+			break;
+		}
+		while ((c = getchar()) != '\n' && c != EOF) { }
```

Comment:
This won't work since `c` is not defined here. Please try to run your code before pushing fixes ;)

Reply:
I will fix it now, and send it again.